prometheus-to-sd/translator/stackdriver.go

```diff
@@ -18,27 +18,49 @@ package translator
 
 import (
 	"fmt"
+	"sync"
 
 	"github.com/golang/glog"
 	v3 "google.golang.org/api/monitoring/v3"
 
 	"k8s.io/contrib/prometheus-to-sd/config"
 )
 
+const (
+	maxTimeseriesesPerRequest = 200
+)
+
 // SendToStackdriver sends http request to Stackdriver to create the given timeserieses.
 func SendToStackdriver(service *v3.Service, config *config.GceConfig, ts []*v3.TimeSeries) {
 	if len(ts) == 0 {
 		glog.Warningf("No metrics to send to Stackdriver")
 		return
 	}
 
-	req := &v3.CreateTimeSeriesRequest{TimeSeries: ts}
 	proj := fmt.Sprintf("projects/%s", config.Project)
 
-	_, err := service.Projects.TimeSeries.Create(proj, req).Do()
-	if err != nil {
-		glog.Errorf("Error while sending request to Stackdriver %v", err)
-	} else {
-		glog.V(4).Infof("Successfully sent %v timeserieses to Stackdriver", len(req.TimeSeries))
+	var wg sync.WaitGroup
+	wg.Add(numberOfBatches(len(ts)))
```

Comment:
You don't need this function: you can just have `wg.Add(1)` right before `go func(...)` below. It'd be less error prone.